source/lib/ze_lib.h

```diff
@@ -34,22 +36,130 @@ namespace ze_lib
         context_t();
         ~context_t();
 
+    //////////////////////////////////////////////////////////////////////////
+    std::string to_string(const ze_result_t result) {
```

Comment:
I really feel like this should be coming in the form of a header being generated at build time, else we have a maintanance nightmare..  something clever with sed or perhaps something formed by Mako.